linkerd/proxy/http/src/h2.rs

```diff
@@ -157,15 +166,18 @@ where
                         )
                         .executor(trace::Executor::new())
                         .handshake(io)
-                        .instrument(info_span!("h2"));
+                        .instrument(trace_span!("handshake"))
+                        .instrument(span.clone());
```

Comment:
Why isn't this necessary? haven't we already entered it?

Reply:
If the `handshake` span is enabled, then the handshake future will be in the outer `h2` span by virtue of `h2` being the parent of `handshake`. However, if the `handshake` span is _not_ enabled (and it's at the TRACE level, so it probably won't be), entering it will do nothing, so the future will not be guaranteed to be in the outer span in that case.